File: external/boost_1.55.0/libs/python/test/voidptr.py
# Copyright Niall Douglas 2005.
# Distributed under the Boost Software License, Version 1.0.
# (See accompanying file LICENSE_1_0.txt or copy at
# http://www.boost.org/LICENSE_1_0.txt)


"""
>>> from voidptr_ext import *


    Check for correct conversion
    
>>> use(get())

    Check that None is converted to a NULL void pointer
    
>>> useany(get())
1
>>> useany(None)
0

    Check that we don't lose type information by converting NULL
    opaque pointers to None
    
>>> assert getnull() is None
>>> useany(getnull())
0

   Check that there is no conversion from integers ...
   
>>> try: use(0)
... except TypeError: pass
... else: print 'expected a TypeError'

   ... and from strings to opaque objects
   
>>> try: use("")
... except TypeError: pass
... else: print 'expected a TypeError'
"""
def run(args = None):
    import sys
    import doctest

    if args is not None:
        sys.argv = args
    return doctest.testmod(sys.modules.get(__name__))
    
if __name__ == '__main__':
    print "running..."
    import sys
    status = run()[0]
    if (status == 0): print "Done."
    sys.exit(status)

File: external/boost_1.55.0/libs/python/test/voidptr.cpp
// Copyright Niall Douglas 2005.
// Distributed under the Boost Software License, Version 1.0.
// (See accompanying file LICENSE_1_0.txt or copy at
// http://www.boost.org/LICENSE_1_0.txt)

# include <boost/python/return_opaque_pointer.hpp>
# include <boost/python/def.hpp>
# include <boost/python/module.hpp>
# include <boost/python/return_value_policy.hpp>

static void *test=(void *) 78;

void *get()
{
    return test;
}

void *getnull()
{
    return 0;
}

void use(void *a)
{
    if(a!=test)
        throw std::runtime_error(std::string("failed"));
}

int useany(void *a)
{
    return a ? 1 : 0;
}


namespace bpl = boost::python;

BOOST_PYTHON_MODULE(voidptr_ext)
{
    bpl::def("get", &::get, bpl::return_value_policy<bpl::return_opaque_pointer>());
    bpl::def("getnull", &::getnull, bpl::return_value_policy<bpl::return_opaque_pointer>());
    bpl::def("use", &::use);
    bpl::def("useany", &::useany);
}
